.github/workflows/main.yml

```diff
@@ -2,8 +2,6 @@ name: Lint, build & test
 
 on:
   push:
-    branches:
```

Comment:
Merged the `on:` sections of "main.yml" and "feature.yml" since they were doing the same thing anyway.